### PR TITLE
FireSim Support for Hetero-Linux + FireSim Cleanup

### PR DESCRIPTION
**Pull request by abejgonzalez**
This PR allows FireSim to boot heterogeneous BOOM + Rocket SoCs in Linux. Additionally, it removes the FireBoom DUT in FireChip and an unused mixin (since BOOM uses a new DCache). Finally, BOOM is bumped.

~~The heterogeneous config currently doesn't have an AGFI created for it. Do we want to create one for it and have it be in Chipyard?~~

Corresponding FireSim PR: https://github.com/firesim/firesim/pull/364

~~Wait until #275 is merged first.~~

### CODE REVIEW

File: generators/firechip/src/main/scala/TargetMixins.scala [hunk 1]
```diff
@@ -9,11 +9,16 @@ import freechips.rocketchip.tilelink._
...
 import freechips.rocketchip.subsystem._
 import freechips.rocketchip.rocket.TracedInstruction
 import firesim.bridges.{TraceOutputTop, DeclockedTracedInstruction}
 
-import midas.targetutils.{ExcludeInstanceAsserts, MemModelAnnotation}
+import midas.targetutils.MemModelAnnotation
+
+import boom.common.BoomTile
+
+import boom.common.BoomTile
```
Comment (jerryz123) [inline generators/firechip/src/main/scala/TargetMixins.scala]: Avoid double import

**Comment by abejgonzalez** [comment]
@a0u @NathanTP It looks like QEMU is breaking the CI with a bad path to the `toolchains` folder? Did this work before?

**Comment by a0u** [comment]
@abejgonzalez: See #278.

**Comment by jerryz123** [comment]
This is also a BOOM bump. Were the recent test images performed with the bumped BOOM?

**Comment by abejgonzalez** [comment]
> This is also a BOOM bump. Were the recent test images generated with the bumped BOOM?

This BOOM bump has been tested with you and Howie's stuff right? So I assumed everything was tested minus `SmallBoom`

**Pull request merged**